src/Arcus.Scripting.ARM/Scripts/Inject-ArmContent.ps1

```diff
@@ -0,0 +1,113 @@
+        #  afterwards character classes and backreferencing to select the optional single or double quotes
+        $fileToInjectPathRegex = [regex] "^(?:FileToInject\s*=\s*)?([`"`']?)(?<File>.*?)\1?$";
+        $fileMatch = $fileToInjectPathRegex.Match($filePart)
+        if ($fileMatch.Success -ne $True){
+            throw "The file part '$filePart' of the injection instruction could not be parsed correctly"
+        }
+
+        $relativePathOfFileToInject = $fileMatch.Groups["File"];
+        $fullPathOfFileToInject = Join-Path (Split-Path $filePath -Parent) $relativePathOfFileToInject
+
+        # Inject content recursively first
+        InjectFile($fullPathOfFileToInject)
+
+        Write-Host "`t Injecting content of $fullPathOfFileToInject into $filePath" 
+
+        $newString = Get-Content -Path $fullPathOfFileToInject -Raw
+
+        # XML declaration can only appear on the first line of an XML document, so remove when injecting
+        $newString = $newString -replace '(<\?xml).+(\?>)(\r)?(\n)?', ""
+
+        # By default: retain double quotes around content-to-inject, if present
+        $surroundContentWithDoubleQuotes = $match.Value.StartsWith('"') -and $match.Value.EndsWith('"')
+
+        if ($instructionParts.Length -gt 1){
+            $optionParts = $instructionParts | select -Skip 1
+
+            if ($optionParts.Contains("ReplaceSpecialChars")){
+                Write-Host "`t Replacing special characters"
+
+                # Replace newline characters with literal equivalents
+                $newString = $newString -replace "`r`n", "\r\n"
+
+                # Replace tabs with spaces
+                $newString = $newString -replace "`t", "    "
+
+                # Replace " with \"
+                $newString = $newString -replace """", "\"""
+            }
+
+            if ($optionParts.Contains("EscapeJson")){
+                Write-Host "`t JSON-escaping file content"
+
+                # Use regex negative lookbehind to replace double quotes not preceded by a backslash with escaped quotes
+                $newString = $newString -replace '(?<!\\)"', '\"'
+            }
+
+
+            if ($optionParts.Contains("InjectAsJsonObject")){
+                try{
+                    # Test if content is valid JSON
+                    ConvertFrom-Json $newString
+
+                    $surroundContentWithDoubleQuotes = $False
+                }
+                catch{
+                    Write-Error "Content to inject cannot be parsed as a JSON object!"
+                }
+            }
+        }
+
+        if ($surroundContentWithDoubleQuotes){
+            Write-Host "`t Surrounding content in double quotes"
+
+            $newString = '"' + $newString + '"'
+        }
+
+        return $newString;
+    }
+
+    $injectionInstructionRegex = [regex] '"?\${(.+)}\$"?';
+    $injectionInstructionRegex.Replace((Get-Content $filePath -Raw), $replaceContentDelegate) | Set-Content $filePath -Encoding UTF8
```

Comment:
Let's not get the content inline and move to variable?

Reply:
Also done. Thanks.